manager/integration/tests/test_node.py

```diff
@@ -1160,3 +1163,37 @@ def test_node_default_disk_labeled(client, core_api, random_disk_path,  reset_de
 
     node = client.by_id_node(cases["unlabeled"])
     assert len(node["disks"]) == 0
+
+
+def test_offline_node():
+    node_worker_label = 'node-role.kubernetes.io/worker'
+
+    k8s_api_client = get_core_api_client()
+    longhorn_api_client = get_longhorn_api_client()
+
+    for node_item in k8s_api_client.list_node().items:
+        if node_worker_label in node_item.metadata.labels and \
+                node_item.metadata.labels[node_worker_label] == 'true':
+            node_external_ip = \
+                node_item.metadata.annotations['rke.cattle.io/external-ip']
+            break
+
+    droplet_id = digitalocean.get_droplet_id(node_external_ip)
+
+    digitalocean.droplet_exec_action(droplet_id,
+                                     action="shutdown",
+                                     expected_status="active",
+                                     desired_status="off")
+    time.sleep(30)
```

Comment:
We need to query the API for the state instead of blindly waiting.

Reply:
I've added 30 seconds wait here just to make sure the shutdown status gets reflected back on k8s, and hence longhorn API.

Reply:
Yeah, that's what I meant. We need to query the API. We don't really know how long it would take for Kubernetes to change state under different circumstances.